pilot/pkg/model/jwks_resolver.go

```diff
@@ -309,6 +309,7 @@ func CreateFakeJwks(jwksURI string) string {
 // Resolve jwks_uri through openID discovery.
 func (r *JwksResolver) resolveJwksURIUsingOpenID(issuer string) (string, error) {
 	// Try to get jwks_uri through OpenID Discovery.
+	issuer = strings.TrimSuffix(issuer, "/")
 	body, err := r.getRemoteContentWithRetry(issuer+openIDDiscoveryCfgURLSuffix, networkFetchRetryCountOnMainFlow)
```

Comment:
maybeshould use url.JoinPath?

Reply:
Actually I am a bit worried about the strategy to append `/.well-known/openid-configuration`. It MUST be at the root afaik? so a user specifying `https://foo.com/bar` should not get `https://foo.com/bar/.well-known/openid-configuration`

Reply:
Per https://openid.net/specs/openid-connect-discovery-1_0.html, it may have use cases as multi-tenant host serving multiple issuers. and https://learn.microsoft.com/en-us/azure/active-directory/develop/v2-protocols-oidc has path specific `/.well-known/openid-configuration`.

Reply:
TIL